site/app/views/course/CourseMaterialsView.php

```diff
@@ -140,13 +142,28 @@ public function listCourseMaterials(array $course_materials_db) {
         ]);
     }
 
+    private function removeEmptyFolders(array &$course_materials): bool {
+        foreach ($course_materials as $path => $course_material) {
+            if (is_array($course_material)) {
+                if ($this->removeEmptyFolders($course_material)) {
+                    unset($course_materials[$path]);
+                }
+            }
+            else {
+                return false;
```

Comment:
Why the `return false;` here? Does `$course_materials` here have an implicit sort that it has all folders first, followed by files? Is it important that we enforce this sort here in some way? Can we just remove the whole `else` branch, and just have it always iterate over all `$course_materials` here? I don't think any instructor has such a long list of course materials that we need to worry about performance optimizations within this loop at the cost of having some implicit necessary structure that is easy to overlook.